src/createStore.js

```diff
@@ -101,6 +102,15 @@ export default function createStore(reducer, initialState) {
     try {
       isDispatching = true;
       currentState = currentReducer(currentState, action);
+
+      if(currentState instanceof StateAndEffect) {
+        var effect = currentState.effect;
+        // Since side effects may dispatch at any time, don't run them
+        // immediately since we don't want cascading dispatches
+        setTimeout(() => effect(dispatch, getState), 0);
```

Comment:
Does each effect need to run in a separate task or could we queue all these up and loop through them at the end of the current dispatch cycle?